src/jit/lowerxarch.cpp

```diff
@@ -2410,6 +2410,50 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
                 break;
         }
     }
+    else if ((intrinsicID >= NI_FMA_MultiplyAdd) && (intrinsicID <= NI_FMA_MultiplySubtractNegatedScalar))
+    {
+        assert(numArgs == 3);
+        assert(op1->OperIsList());
+
+        GenTreeArgList* argList = op1->AsArgList();
+        op1                     = argList->Current();
+
+        argList      = argList->Rest();
+        GenTree* op2 = argList->Current();
+
+        argList      = argList->Rest();
+        GenTree* op3 = argList->Current();
+
+        if (IsContainableHWIntrinsicOp(node, op3))
+        {
+            // 213 form: op1 = (op2 * op1) + [op3]
+            MakeSrcContained(node, op3);
+        }
+        else if (IsContainableHWIntrinsicOp(node, op2))
+        {
+            // 132 form: op1 = (op1 * op3) + [op2]
+            MakeSrcContained(node, op2);
+        }
+        else if (IsContainableHWIntrinsicOp(node, op1))
+        {
+            // Intrinsics with CopyUpperBits semantics cannot have op1 be contained
+
+            if ((flags & HW_Flag_CopyUpperBits) == 0)
+            {
+                // 231 form: op3 = (op2 * op3) + [op1]
+                MakeSrcContained(node, op1);
+            }
+        }
+        else
+        {
+            // TODO-XArch-CQ: Technically any one of the three operands can
```

Comment:
@CarolEidt, will there be any problems if we set op1, op2, and op3 as regOptional? Will the JIT pick just one, or will it actually try to spill all three?

Reply:
You have to pick just one. #6361 is the issue that tracks allowing multiple operands to be specified as regOptional.

Reply:
Fixed.